### ISSUE TITLE
(Thunkable X) Improve the properties panel for IDE

### ISSUE THREAD

**Issue by AlexITMan** [thunkable/thunkable-issues#389]
I suggest to simplify working with properties to simplify the properties panel, remove the simple and Advanced tabs, placing all properties in the expanded form on a single panel. This will eliminate thousands of unnecessary button clicks in order to navigate between the Simple and Advanced tabs, as well as constantly expand properties in collapsed panels. In addition, unnecessary duplication of properties will be eliminated, which creates confusion. How does the Width property in the Simple tab differ from the Width in the Advanced tab, etc.? Why duplicate properties? Why should they be divided into Simple and Advanced, when scrolling and browser search can easily find the desired properties without unnecessary clicks.

**Comment by DLM-thunkable** [comment]
I absolutely agree, this would make the IDE much easier and quicker to use.

**Comment by mark-friedman** [comment]
> Why should they be divided into Simple and Advanced, when scrolling and browser search can easily find the desired properties without unnecessary clicks.

The goal for the existence of the Simple and Advanced tabs was to keep beginning users from being overwhelmed by the choices available to them, i.e. giving them a small set of useful, consistent and easy to understand properties, while allowing advanced users to access a larger set.

Perhaps the compromise solution would be to have a user-level setting for which tab you prefer to appear by default.

**Comment by DLM-thunkable** [comment]
Thanks, I think that would be a great compromise.

**Comment by AlexITMan** [comment]
Mark, selecting the default tab is not a solution to the problem, because the tabs have a different set of properties. For example, the Advanced tab doesn't have a Text property for the button, and how can I change this property if the Advanced tab is displayed by default?

My suggestion is that all the properties of the selected component should be displayed in a single panel. You can add a Checkbox to display additional properties, or you can add all properties to the Advanced tab and open it by default.

Advanced properties will only cause difficulties if there is no clear help for using them. But in this case, I can provide a link to my reference book, where can see documentation for these properties for free.

**Comment by AlexITMan** [comment]
Look at the image below, where two property panels are given for comparison - Thunkable X and Delphi. I don't suggest doing the exact same panel in Thunkable X, but there's a lot to think about.

Do you think that Object Inspector is difficult to work with? Nothing like that. Despite the large amount of information, it is very easy to work with it, even for those who have never programmed. That's why Delphi was very popular at the time.

![изображение](https://user-images.githubusercontent.com/20015046/74730644-9134a400-5257-11ea-8059-1ca863e99044.png)

**Comment by AlexITMan** [comment]
Option

![изображение](https://user-images.githubusercontent.com/20015046/74734547-d78e0100-525f-11ea-8659-aa44214a6f54.png)

**Comment by gre4twh1te** [comment]
> Mark, selecting the default tab is not a solution to the problem, because the tabs have a different set of properties. For example, the Advanced tab doesn't have a Text property for the button, and how can I change this property if the Advanced tab is displayed by default?

I agree that this would have to be adapted as well. It is annoying when you need to switch between the two every time because some settings are here and some over there...
